eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/Iterate.java

```diff
@@ -2423,6 +2423,26 @@ public static boolean notEmpty(Iterable<?> iterable)
         return result == null ? ifNone : result;
     }
 
+    /**
+     * Searches for the first occurrence of object, returns -1 if the object is not found.
+     */
+    public static <T> int indexOf(Iterable<T> iterable, Object object)
+    {
+        if (iterable instanceof ArrayList<?>)
+        {
+            return ArrayListIterate.indexOf((ArrayList<T>) iterable, object);
+        }
+        if (iterable instanceof List<?>)
+        {
+            return ListIterate.indexOf((List<T>) iterable, object);
+        }
+        if (iterable != null)
+        {
+            return IterableIterate.indexOf(iterable, object);
+        }
+        throw new IllegalArgumentException("Cannot perform indexOf on null");
+    }
```

Comment:
I think this is to implement the sorted bag adapter. This must mean that the indexOf method is unsupported everywhere, and you could leave your new implementation unsupported as well if you'd like.

Reply:
while I was at it I thought it would be nice to do it :) should I remove it?